sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/TypeCoercion.scala

```diff
@@ -590,6 +591,22 @@ object TypeCoercion {
   }
 
   /**
+   * Coerces NullTypes of a Stack function to the corresponding column types.
+   */
+  object StackCoercion extends Rule[LogicalPlan] {
+    def apply(plan: LogicalPlan): LogicalPlan = plan resolveExpressions {
+      case s @ Stack(children) if s.childrenResolved =>
+        val schema = s.elementSchema
```

Comment:
Oops. This breaks [GeneratorFunctionSuite](https://github.com/apache/spark/blob/master/sql/core/src/test/scala/org/apache/spark/sql/GeneratorFunctionSuite.scala#L30).

For `StackCoercision`, we need the schema which is based on the number of rows (and derived columns) from the **first argument**. Actually, the content of the first argument. The validate of the value is done by `Stack.checkInputDataTypes`. It seems we cannot add `StackCoercision` in this manner. How do you think about this?